README.md

```diff
@@ -75,4 +75,126 @@ $ pytest .\tests_e2e\
+
+### Virtual environments
+
+This application can also be run within a virtual machine (VM) by using [Vagrant](https://www.vagrantup.com/downloads.html).
+
+Vagrant requires a hypervisor installed. This application is configured to use Windows [Hyper-V](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/about/). 
+
+#### Setting up Hyper-V
+
+1. Install Hyper-V
+2. Add an external switch in Hyper-V Manager as per "Tip 1" in these [instructions](https://techcommunity.microsoft.com/t5/virtualization/vagrant-and-hyper-v-tips-and-tricks/ba-p/382373).
+3. Virtualization needs to be turned on in the firmware. This must be done via the BIOS. This varies by machine but generally the steps are:
+   1. Restart the system
+   2. Press F2 (or alternate shown on screen) to enter BIOS
+   3. Navigate to Advanced
+   4. Select Virtualization Technology and then press the Enter key and select Enabled
+   5. Escape and save
+
+#### Run the application on the VM
+
+To run the VM, run the following command as an admin:
+
+`vagrant up`
+
+You'll be asked to choose a switch to attach to the Hyper-V instance. Select the external switch created during the Hyper-V setup.
+
+The machine will now report it's IP address. Take note of this as it will be used to access the application later.
+
+The command may request your username and password. Enter your Windows credentials (the `@domain` is optional).
+
+The Vagrantfile within this repo will:
+
+- Pull down a Ubuntu Linux image to run the app
+- Prep your VM for Python installation
+- Install Python 3.6.6
+- Install poetry and load any dependencies
+- Launch the TODO app
+  
+Add the IP of the VM to your browser with the port of `5000` to see the TODO application.
+
+##### Example output
+
+````
+Windows PowerShell
+Copyright (C) Microsoft Corporation. All rights reserved.
+
+Try the new cross-platform PowerShell https://aka.ms/pscore6
+
+Hello and welcome to...
+  _____                   _    	   _____ 		       _____ _          _ _ 
+ |  __ \                 ( )      |  __ \                     / ____| |        | | |
+ | |__) |   _  __ _ _ __ |/ ___   | |__) |____      _____ _ __ (___ | |__   ___| | |
+ |  _  / | | |/ _` | '_ \  / __|  |  ___/ _ \ \ /\ / / _ \ '__\___ \| '_ \ / _ \ | |
+ | | \ \ |_| | (_| | | | | \__ \  | |  | (_) \ V  V /  __/ |  ____) | | | |  __/ | |
+ |_|  \_\__, |\__,_|_| |_| |___/  |_|   \___/ \_/\_/ \___|_| |_____/|_| |_|\___|_|_|
+         __/ |                                                                     
+        |___/                                                                      
```

Comment:
Don't think you need all of this example in the README. Especially this part, pretty though it is

Reply:
Taking out the "extra" bit